src/MIDebugEngine/Engine.Impl/Breakpoints.cs

```diff
@@ -111,7 +111,28 @@ private static MIBreakpointState StringToBreakpointState(string addr)
 
             string basename = System.IO.Path.GetFileName(documentName);     // get basename from Windows path
             basename = process.EscapePath(basename);
-            BindResult bindResults = EvalBindResult(await process.MICommandFactory.BreakInsert(basename, line, condition, ResultClass.None), pbreak);
+
+            Checksum[] checksums = null;
+#if CORECLR
+            if (process.MICommandFactory.SupportsBreakpointChecksums())
+            {
+                // TODO: This will need to be configurable in the future somehow
+                // TODO: We might want the MICommandFactory to return the algorithm we should use
+                // TODO: But HashAlgorithmID is not part of MICore becuase it uses AD7Guids
+                HashAlgorithmId[] hashAlgorithmIds = { HashAlgorithmId.SHA1Normalized };
```

Comment:
For 2, I have already added the following code into DocumentChecksum.h in vsdebug. We will need to merge it to update 3. 
```C++
    void SetGuidAlgorithm(REFGUID guidAlgorithm)
    {
        if (IsEqualGUID(guidAlgorithm, guidSourceHashMD5Normalized))
        {
            m_guidAlgorithm = guidSourceHashMD5;
        }
        else if (IsEqualGUID(guidAlgorithm, guidSourceHashSHA1Normalized))
        {
            m_guidAlgorithm = guidSourceHashSHA1;
        }
        else if (IsEqualGUID(guidAlgorithm, guidSourceHashSHA256Normalized))
        {
            m_guidAlgorithm = guidSourceHashSHA256;
        }
        else
        {
            m_guidAlgorithm = guidAlgorithm;
        }
    }
```